lib/GithubApp/index.js

```diff
@@ -0,0 +1,93 @@
+const {join} = require('path');
+const fetch = require('node-fetch');
+const jwt = require('jsonwebtoken');
+const BASE = 'https://api.github.com';
+
+module.exports = class GithubApp {
+    constructor({appId, appPrivateKey} = {}) {
+        this.appId = appId;
+        this.appPrivateKey = appPrivateKey;
+
+        this.getJwtAppToken = this.getJwtAppToken.bind(this);
+        this.getUserToken = this.getUserToken.bind(this);
+    }
+
+    getJwtAppToken() {
+        return jwt.sign(
+            {
+                iss: this.appId,
+            },
+            this.appPrivateKey,
+            {
+                algorithm: 'RS256',
+                expiresIn: '120s',
+            }
+        );
+    }
+
+    headers() {
+        return {
+            'Accept': 'application/json',
+            'Authorization': `Bearer ${this.getJwtAppToken()}`,
+        }
+    }
+
+    async getInstallationId() {
+        const url = join(BASE, 'app/installations');
```

Comment:
I would intorduce another helper called -> 

```
const githubRequest = async(url)  => {
    const quailifiedUrl = join(BASE, url);

        const response = await fetch(
            url,
            {
                headers: this.headers(),
            }
        );

      if (response.ok) {
            return await response.json();
        }

        // eslint-disable-next-line no-console
        console.error(
            [
                `Error in request ${url}`,
                JSON.stringify(await response.json(), null, 2),
            ].join('\n')
        );

        return;
}
```

This way you can utilize it as ->
```
async getInstallationId() {
    const [{ id }] = await githubRequest('app/installations');

  return id;
}
```